lib/services/sf.js

```diff
@@ -782,6 +782,13 @@ function StateAbstract(options)
     requestOptions.headers =
       Util.apply(this.getDefaultReqHeaders(), requestOptions.headers || {});
 
+    if (Util.isLoginRequest(requestOptions.url)) {
+      Util.apply(requestOptions.headers, {
+        "CLIENT_APP_VERSION": requestOptions.json.data.CLIENT_APP_VERSION,
```

Comment:
Does `requestOptions.json.data` always contain `CLIENT_APP_VERSION` and `CLIENT_APP_DATA` for login requests?

Reply:
Yes, in the body, it looks like the body alwasys contains the client version and app_id.